src/Maestro/DependencyUpdateErrorProcessor/DependencyUpdateErrorProcessor.cs

```diff
@@ -0,0 +1,306 @@
+                }
+            }
+            catch (Exception ex)
+            {
+                Logger.LogError(ex.Message, "Could not create a github issue.");
+            }
+        }
+
+        private async Task<GitHubClient> AuthenticateGitHubClient(string issueRepo)
+        {
+            string gitHubToken = null;
+            IGitHubTokenProvider gitHubTokenProvider = Context.GetService<IGitHubTokenProvider>();
+            long installationId = await Context.GetInstallationId(issueRepo);
+            gitHubToken = await gitHubTokenProvider.GetTokenForInstallationAsync(installationId);
+            Logger.LogInformation("GitHub token acquired for " + issueRepo);
+            Octokit.ProductHeaderValue _product;
+            string version = Assembly.GetExecutingAssembly()
+                .GetCustomAttribute<AssemblyInformationalVersionAttribute>()
+                .InformationalVersion;
+            _product = new Octokit.ProductHeaderValue("Maestro", version);
+            var client = new GitHubClient(_product);
+            var token = new Credentials(gitHubToken);
+            client.Credentials = token;
+            return client;
+        }
+
+        private async Task CreateOrUpdateGithubIssueAsync(RepositoryBranchUpdateHistoryEntry repositoryBranchUpdateHistory , string issueRepo)
+        {
+            Logger.LogInformation("Something failed in the repository : " + repositoryBranchUpdateHistory.Repository);
+            string fyiHandles = _options.FyiHandle;
+            IReliableDictionary<(string, string), int> gitHubIssueEvaluator =
+            await _stateManager.GetOrAddAsync<IReliableDictionary<(string,string), int>>("gitHubIssueEvaluator");
+            var client = await AuthenticateGitHubClient(issueRepo);
+            StringBuilder description = new StringBuilder("Something failed during dependency update" +
+                Environment.NewLine + Environment.NewLine);
+            switch (repositoryBranchUpdateHistory.Method)
+            {
+                // Only UpdateAssetAsync method has subscriptionId as one of the parameters in the arguments column. Other methods do not have this info.
+                case "UpdateAssetsAsync":
+                    // append to the description only if the subscription is valid else do not create the issue
+                    if (!String.IsNullOrEmpty(UpdateAsyncArgumentsParse(repositoryBranchUpdateHistory.Arguments)))
+                    {
+                        description.Append(UpdateAsyncArgumentsParse(repositoryBranchUpdateHistory.Arguments));
+                        break;
+                    }
+                    return;
+                // since this is not really an error, skipping this method
+                case "SynchronizePullRequestAsync":
+                    return;
+                // for all the other methods
+                default:
+                    description.Append("Repository :" + repositoryBranchUpdateHistory.Repository);
```

Comment:
Why don't we append the Repository in the UpdateAssetsAsync case?

Reply:
Ooo, it's because "UpdateAsyncArgumentsParse" already wrote those.  That's... a bit tricky to tell, but it's probably fine. Ignore this one.